provider/disk_utils.py

```diff
@@ -0,0 +1,129 @@
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param did: The ID of the VM's disk.
+    :param image_name: The image name. e.g: stg0
+    :param dst_dir: The destination directory for mounting.
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    """
+    session = vm.wait_for_login()
+    is_windows = params.get("os_type") == "windows"
+    image_params = params.object_params(image_name)
+    img_size = image_params.get("image_size")
+
+    if fstype is None:
+        fstype = "ntfs" if is_windows else "xfs"
+
+    if not dst_dir:
+        dst_dir = f"{did}:\\{image_name}" if is_windows else f"/mnt/{image_name}"
+        if is_windows:
+            session.cmd(f"mkdir {dst_dir}")
+        else:
+            session.cmd(f"mkdir -p {dst_dir}")
+
+    if is_windows:
+        utils_disk.update_windows_disk_attributes(session, did)
+        utils_disk.clean_partition_windows(session, did)
+        return utils_disk.configure_empty_disk(session, did, img_size, "windows")[0]
+    else:
+        # Send only the disk ID
+        disk_id = did.split("/")[-1]
+        utils_disk.create_filesyetem_linux(session, disk_id, fstype)
+        if utils_disk.mount(did, dst_dir, fstype, session=session):
+            return dst_dir
+
+
+def execute_io_test(
+    params,
+    vm,
+    image,
+    serial,
+    fstype=None,
+    dst_dir=None,
+    io_command=None,
+    clean=True,
+    ignore_all_errors=False,
+    timeout=60,
+):
+    """
+    Execute the I/O write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param serial: The serial number of the disk
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_command: The command for I/O test.
+    :param clean: If True, cleans the environment
+    :param ignore_all_errors: Checks the errors of the I/O command
+    :param timeout: The timeout for the I/O command.
+    """
+    session = vm.wait_for_login()
+    is_windows = params["os_type"] == "windows"
+
+    did = get_disk_by_serial(params.get("os_type"), serial, session)
+    disk = init_disk_by_id(params, vm, did, image, dst_dir, fstype)
+
+    if not dst_dir:
+        dst_dir = f"{did}:\\{image}" if is_windows else f"/mnt/{image}"
+
+    try:
+        # Start to run I/O command
+        io_cmd = io_command % disk
+        session.cmd(io_cmd, timeout, ignore_all_errors=ignore_all_errors)
+    finally:
+        if clean:
+            # We need to try to clean up and roll back the environment finally.
+            if not is_windows:
+                utils_disk.is_mount(did, dst_dir, fstype=fstype, session=session)
+                utils_disk.umount(did, dst_dir, fstype=fstype, session=session)
```

Comment:
to aviod umount failed , you may consider using force umount directly:
session.cmd(f"umount -fl {disk}")

Reply:
@qingwangrh The `umount` failure from yesterday was related with dst_dir not being initialized, anyways if you consider this direct command would be better I can update it